### PR TITLE
Wireguard fv handshake wait

### PR DESCRIPTION
**Pull request by mikestephen**
## Todos
- [x] Unit tests (full coverage)
- [x] Integration tests Done

## Release Note

```release-note
None required
```

### CODE REVIEW

File: fv/wireguard_test.go [hunk 1]
```diff
@@ -512,18 +512,9 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported", []api
...
 
-			cc.OnFail = func(msg string) {
-				log.WithField("msg", msg).Info("Connectivity Failed after 30seconds")
-				log.Info("Checking connectivity using ping between wls0 and wls1")
-
-				err, errstr := wls[0].SendPacketsTo(wls[1].IP, 5, 100)
-				log.Infof("wls0 to wls1: err=%v, errstr=%s", err, errstr)
-				err, errstr = wls[1].SendPacketsTo(wls[0].IP, 5, 100)
-				log.Infof("wls1 to wls0: err=%v, errstr=%s", err, errstr)
-			}
```
Comment (neiljerram) [inline fv/wireguard_test.go:523]: I presume this is a case of removing debugging that we no longer need?

File: fv/wireguard_test.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ import (
 
 const (
 	wireguardInterfaceNameDefault       = "wireguard.cali"
-	wireguardMTUDefault                 = 1420
+	wireguardMTUDefault                 = 1440
```
Comment (neiljerram) [inline fv/wireguard_test.go:55]: Is it worth a comment to explain this value?  (And perhaps specifically why 1440 is better than 1420?)

File: fv/wireguard_test.go [hunk 3]
```diff
@@ -994,6 +985,30 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported 3-node
 		}
 
 		cc = &connectivity.Checker{}
+
+		// Check felix Wireguards have handshaked.
```
Comment (neiljerram) [inline fv/wireguard_test.go:989]: handshook ? :smile:

Comment (neiljerram) [reply fv/wireguard_test.go:989]: Or is it handshaken?

But seriously, I think "handshaked" is fine here.

File: fv/wireguard_test.go [hunk 4]
```diff
@@ -994,6 +985,30 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported 3-node
...
+		for i, felix := range felixes {
+			var matchers []types.GomegaMatcher
```
Comment (neiljerram) [inline fv/wireguard_test.go:991]: The preparation of `matchers` is unusual here.  I think it becomes `len(felixes)-1` copies of `BeNumerically(">", 0)` - right?  That feels surprisingly heavy, and it also feels strange that the matchers are constructed **inside** the Eventually func that you're testing.

Comment (neiljerram) [reply fv/wireguard_test.go:991]: But I can see that, if it fails, it's nice to see the whole set of handshake numbers together.  So maybe just move the preparation of the matchers slice before the Eventually call - would that work?

**Review by neiljerram — APPROVED** [review]
Basically looks fine, but I commented on a few possible small markups.

File: fv/wireguard_test.go (hunk 1 above)
Comment (mikestephen) [reply fv/wireguard_test.go:523]: Yes, but also this code would mask a failing test. [Here](https://github.com/projectcalico/felix/blob/master/fv/connectivity/conncheck.go#L56) it describes how if `OnFail` is set, `gingko.Fail()` won't be called.
Looking back through the commit history, this code was added for to debug a failing test, and presumably never got removed.

File: fv/wireguard_test.go (hunk 3 above)
Comment (mikestephen) [reply fv/wireguard_test.go:989]: It sounds awful, the way it is I know, I think "Check felix nodes have performed Wireguard handshakes" is better do you? :)

File: fv/wireguard_test.go (hunk 4 above)
Comment (mikestephen) [reply fv/wireguard_test.go:991]: Thanks Neil, I updated to make the code clearer.

File: fv/wireguard_test.go (hunk 3 above)
Comment (neiljerram) [reply fv/wireguard_test.go:989]: Yes, that last suggestion is nicer, thanks!

File: fv/wireguard_test.go (hunk 1 above)
Comment (neiljerram) [reply fv/wireguard_test.go:523]: Good catch.

File: fv/wireguard_test.go (hunk 2 above)
Comment (mikestephen) [reply fv/wireguard_test.go]: Good idea, just added one.

**Pull request merged**